ferry.go

```diff
@@ -1038,3 +1047,36 @@ func (f *Ferry) checkConnectionForBinlogFormat(db *sql.DB) error {
 
 	return nil
 }
+
+func (f *Ferry) checkSourceForeignKeyConstraints() error {
+	for _, table := range f.Tables {
+		query := fmt.Sprintf(`
+		SELECT COUNT(*)
+		FROM information_schema.table_constraints
+		WHERE constraint_type = 'FOREIGN KEY'
+		AND table_schema='%s'
+		AND table_name='%s'
+		`,
+			table.Schema,
+			table.Name,
+		)
+
+		rows, err := f.SourceDB.Query(query)
+		defer rows.Close()
+
+		var count int
+
+		rows.Next()
+		err = rows.Scan(&count)
+
+		if err != nil {
+			return err
+		}
+
+		if count > 0 {
+			return fmt.Errorf("source contains at least 1 foreign key constraint")
```

Comment:
I think it would be helpful to add the table and the constraint in the error message. Is that doable?

Reply:
Addressed in fef68e5